pkg/asset/machines/azure/machines.go

```diff
@@ -127,7 +127,13 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 		ResourceGroup:        rg,
 		NetworkResourceGroup: networkResourceGroup,
 		PublicLoadBalancer:   publicLB,
-	}, nil
+	}
+
+	if platform.IsARO() {
+		spec.ManagedIdentity = ""
```

Comment:
Use a `managedIdentity` variable to populate the `ManagedIdentity` field, rather than clearing out the field after creating the spec.